src/bentoml/_internal/resource.py

```diff
@@ -224,6 +224,9 @@ def from_spec(cls, spec: int | str | list[int | str]) -> list[int]:
 
         try:
             if isinstance(spec, int):
+                if spec == 0:
+                    # NOTE: this is a special case for the default GPU placement.
+                    return [0]
                 if spec < -1:
                     raise ValueError
                 return list(range(spec))
```

Comment:
here we actually allow -1. Should we allow -1? For CUDA_VISIBLE_DEVICES we can set to -1 to indicated that we want to disable gpu. But for our case, if users did specify gpu in resource configuration, that means they want to use GPU. So maybe -1 should also be disallowed?

Reply:
with -1, torch will also respect that as well. I think user might get surprised if we don't have the same behaviour

Reply:
I think we should